### PR TITLE
fix: add usage stats to detect how often users are calling flow.run

### PR DESCRIPTION
**Pull request by CalebTVanDyke**

### CODE REVIEW

File: buildflow/runtime/runner.py [hunk 1]
```diff
@@ -34,6 +41,10 @@ class _ProcessorRef:
     output_ref: type
 
 
+_SESSION_DIR = os.path.join(tempfile.gettempdir(), 'buildflow')
```
Comment (JoshTanke) [inline buildflow/runtime/runner.py:44]: maybe write this to ~/.config/buildflow

Comment (JoshTanke) [reply buildflow/runtime/runner.py:44]: thats where a lot of other frameworks dump stuff

File: buildflow/runtime/runner.py [hunk 2]
```diff
@@ -72,6 +108,17 @@ def __new__(cls, *args, **kwargs):
         return cls._instance
 
     def run(self, num_replicas: int):
+        if self._enable_usage:
+            print(
+                'Usage stats collection is enabled. To disable add the flag: '
+                '`--disable_usage_stats`.')
+            response = requests.post(
+                'https://apis.launchflow.com/buildflow_usage',
+                data=json.dumps({'id': self._session_id}))
```
Comment (JoshTanke) [inline buildflow/runtime/runner.py:117]: optional idea: Maybe write as a Session dataclass object so its easy for us to add other fields later

**Review by JoshTanke — APPROVED** [review]

File: buildflow/runtime/runner.py (hunk 1 above)
Comment (CalebTVanDyke) [reply buildflow/runtime/runner.py]: done.

File: buildflow/runtime/runner.py (hunk 2 above)
Comment (CalebTVanDyke) [reply buildflow/runtime/runner.py]: done.

**Pull request merged**